benchmark: change ts buffer allocation flow

igoropaniuk opened the pull request:
1. Change timestamp buffer allocation/mapping flow
2. Add misc cosmetic fixes

In case if timestamp buffer is allocated in userspace and new register
user memory API is used for its registering in OP-TEE (introduced in
https://github.com/OP-TEE/optee_client/commit/27888d73d156d4862c07effce7c8c2455774768b),
there is no possibility to keep this mapping permanent among different
TEEC_InvokeCommand invocations. All all SHM are automatically unmapped from
OP-TEE VA space after TEEC_InvokeCommand is handled by OP-TEE.

Fixes: https://github.com/OP-TEE/optee_os/issues/1979

`Signed-off-by: Igor Opaniuk <igor.opaniuk@linaro.org>`

jockebech commented on benchmark_aux.c:
```diff
@@ -121,3 +124,28 @@ uint32_t get_cores(void)
+
+void *mmap_paddr(intptr_t paddr, uint64_t size)
+{
+	int   devmem;
```
nit: Single space between int and devmem.

igoropaniuk replied:
fixed

jockebech commented on main.c:
```diff
@@ -184,27 +173,29 @@ static bool init_emitter(FILE *ts_file)
 	if (!yaml_stream_start_event_initialize(&event, YAML_UTF8_ENCODING))
 		ERROR_GOTO(emitter_delete,
-				"Failed to initialize YAML stream start event");
+			"Failed to initialize YAML stream start event");
```
I would probably keep these (here and below) newline/indent changes in a separate commit in this PR.

igoropaniuk replied:
moved to separate commit

jockebech reviewed (commented):
Just some nitpicks, otherwise I think this should work. Btw, do we want to make the buffer size `TS_BUF_SIZE` compile time configurable? 
`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`

igoropaniuk commented:
@jbech-linaro Regarding `TS_BUF_SIZE` I think the best solution here is to make possible to change it's size in run-time (and `benchmark` app will provide proper size when calling PTA to initialize benchmark state), but it will should also be supplied with proper patches in `optee_client` and `linux`

Commit:
benchmark: detect libteec offset

Add detection of an offset of load-time relocatable libteec for correct
translation of addresses into file name/line numbers.

Signed-off-by: Igor Opaniuk <igor.opaniuk@linaro.org>

igoropaniuk commented:
Added detection of an offset of load-time relocatable `libteec` for correct translation of addresses into file name/line numbers by `analyze.py`. @jbech-linaro could I still keep your R-b tag?

Commit:
benchmark: aligment fixes

Add cosmetic fixes of aligment issues in benchmark app

Signed-off-by: Igor Opaniuk <igor.opaniuk@linaro.org>

Commit:
benchmark: change ts buffer allocation flow

1. Change timestamp buffer allocation/mapping flow
2. Add misc cosmetic fixes

In case if timestamp buffer is allocated in userspace and new register
user memory API is used for its registering in OP-TEE (introduced in
https://github.com/OP-TEE/optee_client/commit/27888d73d156d4862c07effce7c8c2455774768b),
there is no possibility to keep this mapping permanent among different
TEEC_InvokeCommand invocations. All all SHM are automatically unmapped from
OP-TEE VA space after TEEC_InvokeCommand is handled by OP-TEE.

Fixes: https://github.com/OP-TEE/optee_os/issues/1979
Signed-off-by: Igor Opaniuk <igor.opaniuk@linaro.org>

igoropaniuk commented:
@jbech-linaro I guess currently it's hard to keep track of changes in this PR (as since you left R-b tag almost all patch has been rewritten), I'll create a new one
